galaxykit/collections.py

```diff
@@ -172,12 +173,41 @@ def delete_collection(client, namespace, collection, version):
     """
     Delete collection version
     """
-    
     if version == None:
         delete_url = f"content/published/v3/collections/{namespace}/{collection}/"
     else:
         delete_url = f"content/published/v3/collections/{namespace}/{collection}/versions/{version}"
 
     return client.delete(delete_url, parse_json=False)
- 
-    
\ No newline at end of file
+
+def delete_collection_in_repository(client, namespace, collection, repository, version):
```

Comment:
The logic all looks good to me, but I'm thinking that `delete_collection_in_repository` might be folded into `delete_collection` above by adding an optional argument to `delete_collection`, with the default value of it being 'published'.

Let me know if you need me to expand on what I mean. `delete_all_collections` looks good to me, good work!

Reply:
But how do you want to do it? The command has already optional parameter version, so two optional parameters dont work. And also, we need to keep backward compatibility. I have only one idea to distinguish between version and repository by parsing it, but this will work only if we are sure that repository name will never be matched as version.

Reply:
ah I see, you mean when it comes to the actual parsing of the CLI invocation that happens in command.py.

What timeline are we looking at for this PR? If possible, I'd like to avoid adding a whole separate function for deleting in different repositories, and our parsing woes would be resolved relatively easily when I have the galaxykit CLI refactor completed and merged in.